paddle/phi/kernels/fusion/onednn/fused_matmul_kernel.cc

```diff
@@ -166,6 +166,16 @@ class FusedMatmulOneDNNHandler
     }
 
     if (residual_data) {
+      // fill 1 in the front of adesc, to make residual ndims to be same as dst
+      // dims
+      int dst_size = out_ddims.size();
+      int origin_size = residual_data->mem_desc().get_ndims();
+      dnnl::memory::dims expanded_dims = residual_data->mem_desc().get_dims();
+      if (origin_size < dst_size) {
+        expanded_dims.insert(expanded_dims.begin(), dst_size - origin_size, 1);
+        residual_data->mem_desc().reshape(expanded_dims);
```

Comment:
maybe `reshaped = residual_data->mem_desc().reshape(expanded_dims);` and then pass it to L191 will be better. Currently, it happen to work because expanded_dims and any layout are being using.

Reply:
yeah, I tired, you are right. It works~ thanks, I will modify.